components/src/dynamo/vllm/engine_generate.py

```
@@ -0,0 +1,196 @@
+logger = logging.getLogger(__name__)
+
+
+def serialize_routed_experts(routed_experts: Any) -> Optional[str]:
+    """Encode routed experts using vLLM's public base64-of-NumPy format."""
+    if routed_experts is None:
+        return None
+    try:
+        buffer = io.BytesIO()
+        np.save(buffer, routed_experts)
+        return base64.b64encode(buffer.getvalue()).decode("ascii")
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)